### DISCUSSION TITLE
Questions regarding this project

### DISCUSSION THREAD

**Discussion by Raj2032** [rwf2/Rocket#2034] [General]
Hi I have some questions regarding this project:
1. Does it allow me to write applications for Windows, Linux and Android as well?
2. If `1` is applicable, then would I get native performance speeds and will it be lightweight?
3. How does it compare to [tauri](https://github.com/tauri-apps/tauri)?

**Comment by Razican** [comment]
This is a framework to create web applications, I guess you could run a web server in a desktop, but this is not the point of Rocket, many other tools are better suited for that.

**Reply by Raj2032** [reply]
Ok thanks mate :)

**Comment by IniterWorker** [comment]
Hi @Raj2032,

Related to your second topic **tauri != rocket.rs**. 

Please, mark the topic as answered.

B. r

**Reply by Raj2032** [reply]
Yep sure.

**Reply by Raj2032** [reply]
Sorry I don't think I can mark it as answer unlike other discussions repo as I don't see an option to do so :/

**Comment by somehowchris** [comment]
@Raj2032 Just to wrap it up what the others intended to say.

-----------------------

`#1` Sure, rocket can build to what ever [rust target](https://doc.rust-lang.org/nightly/rustc/platform-support.html) you choose.  Windows and Linux are on Tier 1 Support, Andriod (and iOS) on Tier 2. This means it should at least should work.

-----------------------

`#2` Rust uses the HIR, MIR, LLVM interpreters to get to your compiled machine code ([this blogpost](https://blog.rust-lang.org/2016/04/19/MIR.html) shows the structure, sry couldn't find a better visualization for it). This should result in a binary about as performant (if not better) as C++. If you need to integrate this into some other code structure like swift or kotlin, it highly depends on what they do with your code or if they just use the ffi of the binary. Rust uses a zero-cost abstraction model and has 3 optimization levels, this means if you write code only what you actually use will be included and it will be optimized the hell (and back 😄 )

-----------------------

`#3` As @IniterWorker and @Razican pointed out. Rocket doesn't really pursue the same goals as tauri. I haven't dealt with tauri but it seems like its a competitor to electron. Such applications are sued to have a frontend and a backend for taking actions which aren't supported in a webview. It seems like tauri has it's own backend setup and therefore has no need for rocket. If you would wish to write your frontend in rust have a look at [yew](https://yew.rs/). 

Rocket itself gives you a web endpoint to handle data, usually Rest APIs, some are rendering templates and forms with a semi to fully stateless client (meaning the client (browser, curl,...) only has the data needed to i.e. edit a form). It seems as frameworks like tauri do not use a full browser setup as you would write apps with stateful clients, both backend (local) and frontend (local) seem to keep some state to let the interaction flow be handled smoothly.

If you need to write an non-local api, rocket would surely be a (if not the) choice to go with.

-----------------------

I hope this clears things up a bit, if you have any other questions, just ask. If not I would like to kindly ask you to mark this discussion as Answered
